src/FlaUI.Core/Tools/Retry.cs

```diff
@@ -56,12 +56,13 @@ public static RetryResult<T> While<T>(Func<T> retryMethod, Func<T, bool> checkMe
                 }
                 catch (Exception ex)
                 {
+                    lastException = ex;
+                    retryResult.SetException(ex);
                     if (!ignoreException)
                     {
+                        retryResult.Finish(lastValueOnTimeout ? lastValue : defaultOnTimeout, false);
```

Comment:
Since we are anyway throwing an exception just afterwards and not returning the `retryResult`, what does this actually improve?

Reply:
In our solution before throwing an exception we log all retry attempts to analyze them later

Reply:
So a logging call or hook would be needed then. Without it, the change in `retryResult` is not used anywhere as it is not returned at all.

Reply:
yes, we use it like: 
 try
 {
    Retry
 }
 finally
 {
     Log retryResult
 }